src/modules/status-list/status-list.controller.ts

```diff
@@ -0,0 +1,173 @@
+  @ApiForbiddenResponse({ description: 'User is not authorized to revoke.' })
+  @ApiConflictResponse({ description: 'Credential already revoked.' })
+  async initiateStatusUpdate(
+    @User() currentUser: string,
+    @Body() { verifiableCredential }: RegisterRevokeInputDto
+  ): Promise<StatusListCredentialDto> {
+    const namespace = verifiableCredential.credentialSubject.role.namespace;
+    const { 2: isAuthorizedRevoker } = await Promise.all([
+      this.roleService.verifyEnrolmentIssuer({
+        issuerDID: DID.from(verifiableCredential.issuer).did,
+        claimType: namespace,
+      }),
+      this.statusListService.verifyCredential(verifiableCredential),
+      this.revocationVerificationService.verifyRevokerAuthority(
+        namespace,
+        currentUser
+      ),
+    ]);
+
+    if (!isAuthorizedRevoker) {
+      throw new ForbiddenException(
+        `${currentUser} is not allowed to revoke ${namespace}`
+      );
+    }
+
+    return await this.statusListService.markStatusListCredential(
+      verifiableCredential,
+      currentUser
+    );
+  }
+
+  @Auth()
+  @Post('/credentials/status/finalize')
+  @ApiCreatedResponse({ type: StatusListVerifiableCredentialDto })
+  @ApiForbiddenResponse({ description: 'User is not authorized to revoke.' })
+  @ApiBadRequestResponse()
+  async persistUpdate(
+    @Body() { statusListCredential }: FinalizeUpdateInputDto
+  ): Promise<StatusListVerifiableCredentialDto> {
+    const issuer = DID.from(statusListCredential.issuer).did;
+    const statusListId = statusListCredential.id;
+
+    const statusList = await this.statusListService.getNamespaceByStatusListId(
+      statusListId
+    );
+
+    if (!statusList) {
+      throw new BadRequestException('Credential was not registered');
+    }
+
+    const { 1: isAuthorizedRevoker } = await Promise.all([
+      this.statusListService.verifyCredential(statusListCredential),
+      this.revocationVerificationService.verifyRevokerAuthority(
+        statusList.namespace,
+        issuer
+      ),
+    ]);
+
+    if (!isAuthorizedRevoker) {
+      throw new ForbiddenException(
+        `${issuer} is not allowed to revoke ${statusList.namespace}`
+      );
+    }
+
+    return await this.statusListService.addSignedStatusListCredential(
+      statusListCredential
+    );
+  }
+
+  @Get('/:credentialId')
+  @ApiOkResponse({ type: StatusListVerifiableCredentialDto })
+  @ApiNoContentResponse({ description: 'Credential not revoked.' })
+  async getStatusListCredential(
+    @Param('credentialId') credentialId: string,
+    @Res() response: Response
+  ): Promise<Response> {
+    const credential = await this.statusListService.getCredential(credentialId);
+    const statusList = await this.statusListService.getStatusList(
+      credential?.entry?.statusListCredential
+    );
+
+    return response
+      .status(statusList?.vc ? HttpStatus.OK : HttpStatus.NO_CONTENT)
+      .send(statusList?.vc || null);
+  }
```

Comment:
Currently, every status list id is pointing here. So the `statusListCredential` is the link to this controller method. In the future it will change from `credentialId` to `statusListId` which will be `uuid` i believe. Maybe it's better to already have it like this :thinking: